sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/context/propagation/OtTracePropagator.scala

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.trace.context.propagation
+
+import org.typelevel.otel4s.baggage.Baggage
+import org.typelevel.otel4s.context.propagation.TextMapGetter
+import org.typelevel.otel4s.context.propagation.TextMapPropagator
+import org.typelevel.otel4s.context.propagation.TextMapUpdater
+import org.typelevel.otel4s.sdk.context.Context
+import org.typelevel.otel4s.sdk.trace.SdkContextKeys
+import org.typelevel.otel4s.trace.SpanContext
+import org.typelevel.otel4s.trace.SpanContext.SpanId
+import org.typelevel.otel4s.trace.SpanContext.TraceId
+import org.typelevel.otel4s.trace.TraceFlags
+import org.typelevel.otel4s.trace.TraceState
+import scodec.bits.ByteVector
+
+import java.util.Locale
+
+private final class OtTracePropagator extends TextMapPropagator[Context] {
+
+  import OtTracePropagator.Headers
+  import OtTracePropagator.Const
+
+  override def fields: List[String] = List(
+    Headers.TraceId,
+    Headers.SpanId,
+    Headers.Sampled
+  )
+
+  override def extract[A: TextMapGetter](ctx: Context, carrier: A): Context = {
+    val spanContext = for {
+      traceIdHex <- TextMapGetter[A].get(carrier, Headers.TraceId)
+      traceId <- ByteVector.fromHex(traceIdHex) if isValidTraceId(traceId)
+
+      spanHexId <- TextMapGetter[A].get(carrier, Headers.SpanId)
+      spanId <- ByteVector.fromHex(spanHexId) if isValidSpanId(spanId)
+
+      sampled <- TextMapGetter[A]
+        .get(carrier, Headers.Sampled)
+        .flatMap(_.toBooleanOption)
+
+    } yield SpanContext(
+      traceId = traceId,
```

Comment:
The Open Trace `traceId` might be shorter than 16 bytes. So we should pad the value:
```suggestion
      traceId = padLeft(traceId, SpanContext.TraceId.Bytes.toLong),
```

```scala
private def padLeft(input: ByteVector, expectedSize: Long): ByteVector =
  if (input.size <= expectedSize) input.padLeft(expectedSize) else input
```

Reply:
And we can implement a similar test: https://github.com/open-telemetry/opentelemetry-java/blob/main/extensions/trace-propagators/src/test/java/io/opentelemetry/extension/trace/propagation/OtTracePropagatorTest.java#L226-L236

Reply:
Done